tests/test_kernels.py

```diff
@@ -17,57 +21,67 @@
+    return mask_data
+
+
+def _make_irregular_grid_data(shape: Tuple[int, int]) -> np.ndarray:
+    return 0.5 + rng.random(shape)
+
+
 ################## Scalar Laplacian tests ##############################################
 @pytest.fixture(scope="module", params=scalar_grids)
 def grid_type_field_and_extra_kwargs(request):
     grid_type = request.param
-    ny, nx = (128, 256)
-    data = np.random.rand(ny, nx)
+    shape = (128, 256)
+
+    data = _make_random_data(shape)
 
     extra_kwargs = {}
-    if grid_type == GridType.REGULAR_AREA_WEIGHTED:
-        area = 0.5 + np.random.rand(ny, nx)
-        extra_kwargs["area"] = area
-    if grid_type == GridType.REGULAR_WITH_LAND:
-        mask_data = np.ones_like(data)
-        mask_data[: (ny // 2), : (nx // 2)] = 0
-        extra_kwargs["wet_mask"] = mask_data
-    if grid_type == GridType.REGULAR_WITH_LAND_AREA_WEIGHTED:
-        area = 0.5 + np.random.rand(ny, nx)
-        extra_kwargs["area"] = area
-        mask_data = np.ones_like(data)
-        mask_data[: (ny // 2), : (nx // 2)] = 0
-        extra_kwargs["wet_mask"] = mask_data
-    if grid_type == GridType.IRREGULAR_WITH_LAND:
-        mask_data = np.ones_like(data)
-        mask_data[: (ny // 2), : (nx // 2)] = 0
-        extra_kwargs["wet_mask"] = mask_data
-        grid_data = 0.5 + np.random.rand(ny, nx)
-        extra_kwargs["dxw"] = grid_data
-        extra_kwargs["dyw"] = grid_data
-        extra_kwargs["dxs"] = grid_data
-        extra_kwargs["dys"] = grid_data
-        extra_kwargs["area"] = grid_data * grid_data
-        extra_kwargs["kappa_w"] = np.ones_like(data)
-        extra_kwargs["kappa_s"] = np.ones_like(data)
-    if grid_type == GridType.TRIPOLAR_REGULAR_WITH_LAND_AREA_WEIGHTED:
-        area = 0.5 + np.random.rand(ny, nx)
-        extra_kwargs["area"] = area
-        mask_data = np.ones_like(data)
-        mask_data[: (ny // 2), : (nx // 2)] = 0
-        mask_data[0, :] = 0  #  Antarctica
-        extra_kwargs["wet_mask"] = mask_data
+    for name in _grid_kwargs[grid_type]:
+        if name == "wet_mask":
+            extra_kwargs[name] = _make_mask_data(shape)
+        elif "kappa" in name:
+            extra_kwargs[name] = np.ones(shape)
+        else:
+            extra_kwargs[name] = _make_irregular_grid_data(shape)
+
+    # special case for POP
     if grid_type == GridType.TRIPOLAR_POP_WITH_LAND:
-        mask_data = np.ones_like(data)
-        mask_data[: (ny // 2), : (nx // 2)] = 0
-        mask_data[0, :] = 0  #  Antarctica
-        extra_kwargs["wet_mask"] = mask_data
-        grid_data = 0.5 + np.random.rand(ny, nx)
-        extra_kwargs["dxe"] = grid_data
-        extra_kwargs["dye"] = grid_data
-        extra_kwargs["dxn"] = grid_data
-        extra_kwargs["dyn"] = grid_data
-        extra_kwargs["tarea"] = grid_data * grid_data
+        # is this important? If not, remove special case
+        extra_kwargs["tarea"] = extra_kwargs["dxe"] * extra_kwargs["dye"]
```

Comment:
Does the POP kernel (or other kernels for that matter) care about consistency between the grid variables?

Reply:
No, none of the kernels cares about this. So I removed this in #80.